impl-base/src/main/java/org/jboss/shrinkwrap/impl/base/URLPackageScanner.java

```diff
@@ -156,7 +156,8 @@ private void handle(File file, String packageName) throws ClassNotFoundException
                 String className = packagePrefix + child.getName().substring(0, child.getName().lastIndexOf(SUFFIX_CLASS));
                 foundClass(className, prefix + className.replace( '.', '/' ) + SUFFIX_CLASS );
             } else if (child.isDirectory() && addRecursively) {
-                handle(child, packageName + "." + child.getName());
+                final String nextPackageName = packageName.length() > 0 ? packageName + "." + child.getName() : child.getName();
```

Comment:
While writing the test I realized that I might be misusing `.addPackages()` by just passing an empty string as the `packages` vararg parameter.
This way everything is scanned.

Do you think this is a valid use case (meaning my fix to the scanner is still valid) or do you think that instead there should be a validation to prevent this?
The single-package parameter variants (e.g. `.addPackage(String)`) do check for null, but not for empty.

I am inclined to allow empty package names but I am not sure...
WDYT?